src/swarmpal/toolboxes/tfa/tfa_processor.py

```diff
@@ -15,192 +15,662 @@
 from swarmpal.toolboxes.tfa import tfalib
 
 
-class TfaMagInputs(ExternalData):
+class TfaInput(ExternalData):
+    """
+    This extends the ExternalData class, so it uses the same inputs, plus a
+    few more to taylor the data to the ones that are being used by the TFA
+    tool.
+
+    Parameters
+    ----------
+    collection : str
+        One of ExternalData.COLLECTIONS
+    start_time : datetime
+        The starting time for the analysis
+    end_time : datetime
+        The ending time for the analysis
+    initialise : bool
+        Set to True to download the requested data. If False, the TfaInput
+        onject will be initialized, but without any actual data inside
+    varname : str
+        The name of the variable to be analysed, e.g. "F" for the magnetic
+        field magnitude by the ASM instrument, or "B_NEC" for the vector
+        field from the VFM instrument, etc.
+    sampling_time : str
+        String that specifies the time between two consecutive measurements
+        to be given according to the ISO 8601 standard for time intervals
+        e.g. PT1S for one second resolution, PT0.02S for 0.02 seconds,
+        PT1M for one minute resolution etc
+    remove_chaos : bool
+        Set to True to allow a subsequent call of subtract_chaos() to
+        remove the CHAOS-Core and CHAOS-Static field model from the data.
+        It only applies to magnetic field variables.
+    """
+
+    # default values for the class parameters
+    TIME_LIMS = []
+    COLLECTION = "SW_OPER_MAGB_LR_1B"
+    VARNAME = "B_NEC"
+    SAMPLING_TIME = 1
+    FLAGNAME = "Flags_B"
+    MAX_FLAG_VAL = 30
```

Comment:
This default seems overly broad. To only get nominal data, Flags_B should be 0 or 1 (for Swarm Alpha, Bravo), or <=9 for Charlie. This flag cleaning should be revisited using the new more general [`.add_filter`](https://viresclient.readthedocs.io/en/latest/api.html#viresclient.SwarmRequest.add_filter), and a range of defaults applied depending on collection name.

Could consider a more general "preprocessing" tool at the PalData level.